subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedCustomTaskExecutionIntegrationTest.groovy

```diff
@@ -590,6 +590,104 @@ class CachedCustomTaskExecutionIntegrationTest extends AbstractIntegrationSpec i
         output.contains "Not caching task ':customTask' because no valid cache key was generated"
     }
 
+    def "task stays up-to-date after loaded from cache"() {
+        file("input.txt").text = "input"
+        buildFile << defineProducerTask()
+
+        withBuildCache().succeeds "producer"
+
+        when:
+        cleanBuildDir()
+        withBuildCache().succeeds "producer"
+        then:
+        skippedTasks as List == [":producer"]
+
+        when:
+        succeeds "producer"
```

Comment:
❌ If you change this line to `withBuildCache().succeeds "producer", "--info"`, then the output will contain:
```
Caching disabled for task ':producer': Gradle does not know how file 'build/empty-dir' was created (output property 'emptyDir'). Task output caching requires exclusive access to output paths to guarantee correctness.
```
As I said otherwise, the root directory needs to be part of the output snapshot.